tribe-ext-disallow-past-starting-dates.php

```diff
@@ -0,0 +1,517 @@
+					border-color: #dc3232 !important;
+					box-shadow: 0 0 2px rgba(204, 0, 0, 0.8) !important;
+				}
+			</style>
+			<?php
+		}
+
+		/**
+		 * Register this extension's asset(s).
+		 */
+		public function register_assets() {
+			$resources_url = trailingslashit( plugin_dir_url( __FILE__ ) ) . 'src/resources/';
+
+			$js = $resources_url . 'js/script.js';
+
+			// `tribe-events-admin` dependency so the `tribe_datepicker_opts` JS variable is set by the time we need to extend it
+			// which comes from /wp-content/plugins/the-events-calendar/src/resources/js/events-admin.js
+			wp_register_script( $this->handle, $js, array(
+				'jquery',
+				'tribe-events-admin'
+			), $this->get_version(), true );
+		}
+
+		/**
+		 * Get the datepicker format from TEC settings. Default to 'Y-m-d'.
+		 *
+		 * @return string
+		 */
+		private function get_datepicker_format() {
+			$datepicker_format = Tribe__Date_Utils::datepicker_formats( tribe_get_option( 'datepickerFormat', 'Y-m-d' ) );
+
+			return $datepicker_format;
+		}
+
+		/**
+		 * Build the $this->script_vars array.
+		 *
+		 * @see wp_localize_script()
+		 *
+		 * @param int $post_id
+		 */
+		private function build_script_vars( $post_id = 0 ) {
+			$this->script_vars['error_class'] = $this->get_error_css_class();
+
+			// Must run $this->get_min_date() before $this->get_max_date()
+			$this->script_vars['min_date'] = $this->get_min_date( $post_id );
+			$this->script_vars['max_date'] = $this->get_max_date( $post_id );
+		}
+
+		/**
+		 * Get the time zone to use for date calculations.
+		 *
+		 * @param int $post_id
+		 *
+		 * @return string
+		 */
+		private function get_time_zone_string( $post_id = 0 ) {
+			$time_zone = Tribe__Events__Timezones::get_event_timezone_string( $post_id );
+
+			/**
+			 * Override the time zone used for date calculations.
+			 *
+			 * @param string $time_zone A named time zone (not manual UTC offset).
+			 * @param int $post_id The Post ID.
+			 *
+			 * @return string
+			 */
+			$time_zone = (string) apply_filters( $this->get_handle_underscores() . '_time_zone', $time_zone, $post_id );
+
+			if ( ! in_array( $time_zone, timezone_identifiers_list() ) ) {
+				// This will fallback to UTC but may also return a TZ environment variable (e.g. EST), which could cause an error for DateTimeZone().
+				$time_zone = date_default_timezone_get();
+			}
+
+			return $time_zone;
+		}
+
+		/**
+		 * Get the PHP DateTime object of "today at midnight" (the first second
+		 * of today) or midnight of a given timestamp.
+		 *
+		 * Because the datepicker is separate from the timepicker, we need to
+		 * make sure we are using midnight whenever setting the script's minDate
+		 * or maxDate.
+		 *
+		 * @param int $post_id
+		 * @param bool|int $timestamp
+		 *
+		 * @return DateTime
+		 */
+		private function get_midnight_datetime( $post_id = 0, $timestamp = false ) {
+			$tz_string = $this->get_time_zone_string( $post_id );
+
+			$time_zone = new DateTimeZone( $tz_string );
```

Comment:
I'd recommend we wrap from here up to the setTime() call in a try {} block (or else do so for the code that calls this method).

Reply:
done